clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -689,6 +689,9 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
                     parsedRecords = new PartitionRecords<>(fetchOffset, tp, parsed);
                     ConsumerRecord<K, V> record = parsed.get(parsed.size() - 1);
                     this.sensors.recordsFetchLag.record(partition.highWatermark - record.offset());
+                } else if (partition.records.iterator().hasNext()) {
```

Comment:
Good point. This is a little subtle, but it doesn't seem incorrect. Since we know that the fetched offset matches our current position, the fetch lag should be in sync with the position. In fact, it may be more intuitive, if we use `fetchOffset` instead of `position`.